### ISSUE TITLE
Warning in HomeAssistant Log

### ISSUE THREAD

**Issue by NicoDreamzZ** [gluap/pyduofern-hacs#51]
Hey, i discovered the following warning in the HA Logs after restarting. Two Rademacher RolloTron devices are connected.

`Detected code that calls async_forward_entry_setup for integration duofern with title: duofern and entry_id: 5ceeb6209d6d91bcffef2669250f657d, during setup without awaiting async_forward_entry_setup, which can cause the setup lock to be released before the setup is done. This will stop working in Home Assistant 2025.1. Please report this issue.`

**Comment by Dondo11** [comment]
I'm seeing this, too.

Are there any plans for fixing this?  It would be a pity to not be able to upgrade HA to 2025.1 and beyond or loose the Rademacher Duofern integration 🥲

**Comment by gluap** [comment]
![Screenshot_20240929-231713.png](https://github.com/user-attachments/assets/d8c2958d-398a-48ed-8811-92460bed22ad)

Seems we have been given yet another 6 extra months to fix it. And this isn't the first extension of the deadline.

**Comment by Dondo11** [comment]
Small sigh of relief 🙂

Just out of interest: How much of an effort would it be to implement the suggested remedy?

**Comment by NicoDreamzZ** [comment]
In my homeassistant instance it sill says deprecated at 2025.1... Maybe this bit can help? https://developers.home-assistant.io/blog/2024/06/12/async_forward_entry_setups/ 

I am not very familiar with python, i think something needs to be changes here:
https://github.com/gluap/pyduofern-hacs/blob/a27915fb87080e2ed8384d78f84a36b98ba7da7f/custom_components/duofern/__init__.py#L132C1-L139C16

**Comment by gluap** [comment]
@NicoDreamzZ which homeassistant version are you using? I just checked and on latest I still get 2025.6 ![Screenshot_20241114-064716.png](https://github.com/user-attachments/assets/3240d4f2-15aa-42f2-b6a2-1a83f4491948)

**Comment by Dondo11** [comment]
I‘m currently on 2024.11.0, skipping every other update unless something of interest was being introduced.Am 14.11.2024 um 06:49 schrieb gluap ***@***.***>:﻿
@NicoDreamzZ which homeassistant version are you using? I just checked and on latest I still get 2025.6 Screenshot_20241114-064716.png (view on web)

**Comment by gluap** [comment]
If you like you could try the latest pyduofern prerelease, it should not have the warning any more. Once it's aged a bit and people have used it I'll upgrade it to a proper release.

**Comment by NicoDreamzZ** [comment]
Shure, i downloaded the latest prerelease and did a complete reboot of my instance. All devices are working fine, and the warning is gone.

**Comment by Dondo11** [comment]
> If you like you could try the latest pyduofern-hacs prerelease …

Dumb question from an uninitiated HA user:  Is there a straightforward way to install that prerelease via the HACS integration?

**Comment by NicoDreamzZ** [comment]
> > If you like you could try the latest pyduofern-hacs prerelease …
> 
> Dumb question from an uninitiated HA user: Is there a straightforward way to install that prerelease via the HACS integration?

Yes, you can select the integration in hacs, tap the three dots in the upper right corner, select Redownload and then on the Dialoge you can select your desired version.

**Comment by Dondo11** [comment]
> Yes, you can select the integration in hacs, tap the three dots in the upper right corner, select Redownload and then on the Dialoge you can select your desired version.

That went without a hitch so far. The warning's gone.  Will see how it goes...

**Comment by Dondo11** [comment]
Happy New Year!

Also happy to report that the prerelease is doing fine so far.

Just upgraded to HA 2025.1.0 and found a new deprecation warning in the system log.  Thought I‘d let you know…

Logger: homeassistant.components.light
Source: helpers/deprecation.py:222
integration: Light (documentation, issues)
First occurred: 11:02:58 (2 occurrences)
Last logged: 11:02:58

SUPPORT_BRIGHTNESS was used from duofern, this is a deprecated constant which will be removed in HA Core 2026.1. Use supported_color_modes instead, please report it to the author of the 'duofern' custom integration

**Comment by gluap** [comment]
THis additional warning is now also fixed.

**Comment by Dondo11** [comment]
Thank you very much for keeping this going!! 🙂👍

**Comment by gluap** [comment]
@bcutter indeed this was still marked as beta. Should now be available as stable. Thank you for the nudge.

**Comment by Dondo11** [comment]
Thanks for the update!  Version 0.5.25 behaving nicely 🙂
